src/lib/state/Store.ts

```diff
@@ -424,15 +424,51 @@ class GlobalStore {
         }
     }
 
-    addToastNotification(toast: ToastMessage, sound?: Sounds) {
-        let toasts = get(this.state.toasts)
-        let id = uuidv4()
-        let timeout = setTimeout(() => {
-            this.removeToast(id)
-        }, toast.remaining_time * 1000)
-        this.state.toasts.set({ ...toasts, [id]: [toast, timeout] })
-        if (sound) {
-            playSound(sound)
+    async addToastNotification(toast: ToastMessage, options?: { sound?: Sounds; context?: NotificationType }) {
+        let allowed = get(SettingsStore.state).notifications.enabled
+        if (options?.context) {
+            switch (options?.context) {
+                case NotificationType.Friends: {
+                    allowed = get(SettingsStore.state).notifications.friends
+                    break
+                }
+                case NotificationType.Messages: {
+                    allowed = get(SettingsStore.state).notifications.messages
+                    break
+                }
+                case NotificationType.Settings: {
+                    allowed = get(SettingsStore.state).notifications.settings
+                    break
+                }
+            }
+        }
+        if (!allowed) return
+        allowed = Notification.permission == "granted"
```

Comment:
Could we let those strings in a const variable?

Reply:
its already predefined in the docs so i dont really see a need for that